executor/src/main/java/org/k8loud/executor/service/MapperServiceImpl.java

```diff
@@ -24,7 +24,8 @@ public Action map(ExecutionRQ executionRQ) {
         Action action = null;
         try {
             Map<String, String> params = executionRQ.getParams();
-            action = (Action) actionClass.getConstructor(Map.class).newInstance(params);
+            if (actionClass != null & params != null)
```

Comment:
null check for:

- actionClass should be removed, it's handled in the catch; logging will be ommited if left like this
- params should be enhanced with logging; maybe it could be removed with usage of lombok's `@NonNull`?

Reply:
moving back with this code. Params should be validated in action and if needed, exception should be thrown